api/src/com/cloud/network/vpc/OSPFZoneConfig.java

```diff
@@ -0,0 +1,329 @@
+        this.password = details.get(Params.PASSWORD.name()) == null ? "" : details.get(Params.PASSWORD.name());
+        this.superCIDRList = details.get(Params.SUPER_CIDR.name()) == null ? new CIDR[0] : NetUtils.convertToCIDR(details.get(Params.SUPER_CIDR.name()).split(","));
+        this.enabled = details.get(Params.ENABLED.name()) == null ? Boolean.FALSE : Boolean.valueOf(details.get(Params.ENABLED.name()));
+    }
+
+    public void setValues(final String protocol, final String ospfArea, final Short helloInterval, final Short deadInterval, final Short retransmitInterval,
+            final Short transitDelay, final String authentication, final String quaggaPassword, final String superCIDRList, final Boolean enabled) throws BadCIDRException {
+        if (protocol != null) {
+            if ((Protocol.OSPF.name().equals(protocol))) {
+                this.protocol = Protocol.valueOf(protocol);
+            } else {
+                throw new InvalidParameterValueException("The quagga protocal can have values of OSPF only " + protocol);
+            }
+        }
+        if (ospfArea != null) {
+            this.ospfArea = ospfArea;
+        }
+        if (helloInterval != null) {
+            this.helloInterval = helloInterval;
+        }
+        if (deadInterval != null) {
+            this.deadInterval = deadInterval;
+        }
+        if (retransmitInterval != null) {
+            this.retransmitInterval = retransmitInterval;
+        }
+        if (transitDelay != null) {
+            this.transitDelay = transitDelay;
+        }
+        if (authentication != null) {
+            if (Authentication.MD5.name().equals(authentication) || Authentication.PLAIN_TEXT.name().equals(authentication)) {
+                this.authentication = Authentication.valueOf(authentication);
+            } else {
+                throw new InvalidParameterValueException("The quagga authentication can have values of MD5 or PLAIN_TEXT and not " + protocol);
+            }
+        }
+        if (quaggaPassword != null) {
+            this.password = quaggaPassword;
+        }
+        if (superCIDRList != null) {
+            String[] cidr_list = superCIDRList.split(",");
+            for (String cidr : cidr_list) {
+                if (!NetUtils.isValidCIDR(cidr)) {
+                    throw new InvalidParameterValueException("The super CIDR is not a valid cidr " + cidr);
+                }
+            }
+            CIDR[] v_cidr = NetUtils.convertToCIDR(cidr_list);
+            if (!NetUtils.cidrListConsistency(v_cidr)) {
+                throw new InvalidParameterValueException("The cidr list is not consistent " + Arrays.toString(cidr_list));
+            }
+            this.superCIDRList = v_cidr;
+        }
+        if (enabled != null) {
+            this.enabled = enabled;
+        }
+    }
+
+    public void setValues(Map<String, String> details) throws BadCIDRException {
```

Comment:
setDefaultValues bootstraps the config. I have a feeling that to make the API easy on the user, I have created an unnecessary complexity. Like I assume default values when not provided. Instead this method could have been much cleaner without any assumptions of default and ensuring that all the values are provided all the time.